### ISSUE TITLE
Problems in training customized datasets

### ISSUE THREAD

**Issue by AbandonedWarlord** [FuyaLuo/PearlGAN#5]
Hi author, I read your paper and found the innovations in it to be great, while trying to train a customized dataset with the code you provided, the fake images generated during the testing phase were all pure black, my training set was structured as follows:
├── my_datasets
          trainA (daytime RGB images)
          trainB (nighttime IR images)
          testA (testing daytime RGB images)
          testB (testing nighttime IR images)
   
├── my_IR_edge_map
├── my_VIS_edge_map

I hope you can help point out where I went wrong in my training, I'd appreciate it!

**Comment by FuyaLuo** [comment]
> Hi author, I read your paper and found the innovations in it to be great, while trying to train a customized dataset with the code you provided, the fake images generated during the testing phase were all pure black, my training set was structured as follows: ├── my_datasets trainA (daytime RGB images) trainB (nighttime IR images) testA (testing daytime RGB images) testB (testing nighttime IR images)
> 
> ├── my_IR_edge_map ├── my_VIS_edge_map
> 
> I hope you can help point out where I went wrong in my training, I'd appreciate it!

Are the generated images from your training phase normal? If it is normal, the test phase needs to check that your test images are entered correctly, and then the test directory should contain only the following two folders:
├── test0 (empty folder)
├── test1 (testing nighttime IR images)

**Comment by AbandonedWarlord** [comment]
> > Hi author, I read your paper and found the innovations in it to be great, while trying to train a customized dataset with the code you provided, the fake images generated during the testing phase were all pure black, my training set was structured as follows: ├── my_datasets trainA (daytime RGB images) trainB (nighttime IR images) testA (testing daytime RGB images) testB (testing nighttime IR images)
> > ├── my_IR_edge_map ├── my_VIS_edge_map
> > I hope you can help point out where I went wrong in my training, I'd appreciate it!
> 
> Are the generated images from your training phase normal? If it is normal, the test phase needs to check that your test images are entered correctly, and then the test directory should contain only the following two folders: ├── test0 (empty folder) ├── test1 (testing nighttime IR images)

Hi, my test set is fine, I just noticed that the gradient explodes at the 16th epoch, the loss is nan.My collective amount of data is very small, about 500 sheets, what parameters do I need to debug to prevent the gradient from exploding

**Comment by FuyaLuo** [comment]
> > > Hi author, I read your paper and found the innovations in it to be great, while trying to train a customized dataset with the code you provided, the fake images generated during the testing phase were all pure black, my training set was structured as follows: ├── my_datasets trainA (daytime RGB images) trainB (nighttime IR images) testA (testing daytime RGB images) testB (testing nighttime IR images)
> > > ├── my_IR_edge_map ├── my_VIS_edge_map
> > > I hope you can help point out where I went wrong in my training, I'd appreciate it!
> > 
> > 
> > Are the generated images from your training phase normal? If it is normal, the test phase needs to check that your test images are entered correctly, and then the test directory should contain only the following two folders: ├── test0 (empty folder) ├── test1 (testing nighttime IR images)
> 
> Hi, my test set is fine, I just noticed that the gradient explodes at the 16th epoch, the loss is nan.My collective amount of data is very small, about 500 sheets, what parameters do I need to debug to prevent the gradient from exploding

I suggest you train with the baseline model (i.e., ToDayGAN) once to see the performance, and then add the losses designed in the PearlGAN model one by one.
